WI #557: Normative Requirements Refinement - Part 2 (#798)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/commitmentdiscountunit.md

The CommitmentDiscountUnit column adheres to the following requirements:

* CommitmentDiscountUnit MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports *commitment discounts*.
* CommitmentDiscountUnit MUST be of type String.
* CommitmentDiscountUnit MUST conform to [StringHandling](#stringhandling) requirements.
* CommitmentDiscountUnit SHOULD conform to [UnitFormat](#unitformat) requirements.
* CommitmentDiscountUnit nullability is defined as follows:

>   * When ChargeCategory is "Usage" or "Purchase" and [CommitmentDiscountId](#commitmentdiscountid) is not null, CommitmentDiscountUnit adheres to the following additional requirements:
>     * CommitmentDiscountUnit MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction".
>     * CommitmentDiscountUnit MAY be null when ChargeClass is "Correction".
>   * CommitmentDiscountUnit MUST be null in all other cases.

Author: In the specification, we currently have 3 pairs of quantity and corresponding unit columns, and in all three cases, the nullability requirements for both quantity and unit are specified identically. As a result, in the case of a correction record, **we allow a null unit regardless of what is in the quantity column, and vice versa**. 

While discussing correction handling, **we confirmed** that it would be more accurate to specify that the **unit must not be null if the corresponding quantity is not null, and vice versa**. This should apply regardless of whether it’s a non-correction or correction record.

```suggestion
  * CommitmentDiscountUnit MUST be null when CommitmentDiscountQuantity is null.
  * CommitmentDiscountUnit MUST NOT be null when CommitmentDiscountQuantity is not null.
```

Author: In addition to here provided thumbs-up, since this was already discussed and approved by TF-1, I’m resolving this conversation.
